src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTest.cs

```diff
@@ -637,5 +637,46 @@ public void Process_Environment()
             }
             );
         }
+
+        [Fact]
+        public void Process_StartInfo()
+        {
+            {
+                Process process = CreateProcessInfinite();
+                process.Start();
+
+                Assert.Equal(ProcessName, process.StartInfo.FileName);
+
+                process.Kill();
+                process.WaitForExit(WaitInMS);
+            }
+
+            {
+                Process process = CreateProcessInfinite();
+                process.Start();
+
+                Assert.Throws<System.InvalidOperationException>(() => (process.StartInfo = new ProcessStartInfo()));
+
+                process.Kill();
+                process.WaitForExit(WaitInMS);
+            }
+
+            {
+                Process process = new Process();
+                process.StartInfo = new ProcessStartInfo(ProcessName);
+                Assert.Equal(ProcessName, process.StartInfo.FileName);
+            }
+
+            {
+                Process process = Process.GetCurrentProcess();
+                Assert.Throws<System.InvalidOperationException>(() => process.StartInfo);
+            }
+
+            {
+                Process process = Process.Start(ProcessName);
```

Comment:
I see.  What's the purpose then of the next line then?  Won't that effectively turn it into the first case, by Close'ing the handle and then refreshing/recreating everything?

Regardless, I see your goal here, but I don't think it's necessary, especially since you'll need to handle the case I mentioned about weeding out the problematic environment variables.